app/controllers/v1/sessions_controller.rb

```diff
@@ -385,15 +385,26 @@ def after_login_actions
     def log_persisted_session_and_warnings
       obscure_token = Session.obscure_token(@session_object.token)
       Rails.logger.info("Logged in user with id #{@session_object&.uuid}, token #{obscure_token}")
-      # We want to log when SSNs do not match between MVI and SAML Identity. And might take future
-      # action if this appears to be happening frequently.
-      if current_user.ssn_mismatch?
-        additional_context = StringHelpers.heuristics(current_user.identity.ssn, current_user.ssn_mpi)
-        log_message_to_sentry(
-          'SessionsController version:v1 message:SSN from MPI Lookup does not match UserIdentity cache',
-          :warn,
-          identity_compared_with_mpi: additional_context
-        )
+      id_mismatch_validations
+    end
+
+    def id_mismatch_validations
+      return unless current_user.loa3?
+
+      check_id_mismatch(@current_user.ssn, @current_user.ssn_mpi, :identity_mpi_conflict_ssn)
```

Comment:
- No need for the `@` here, we can just refer to it as `current_user`  
- Also, can we make these explicitly from `identity`, so `current_user.identity.ssn`? Just want to make sure to do that, otherwise some of these might fall back to MPI anyways and make a useless comparison (I think it is important to log this diff when the `identity` value is nil)